Doc/reference/expressions.rst

```diff
@@ -83,17 +83,78 @@ exception.
    pair: name; mangling
    pair: private; names
 
-**Private name mangling:** When an identifier that textually occurs in a class
-definition begins with two or more underscore characters and does not end in two
-or more underscores, it is considered a :dfn:`private name` of that class.
-Private names are transformed to a longer form before code is generated for
-them.  The transformation inserts the class name, with leading underscores
-removed and a single underscore inserted, in front of the name.  For example,
-the identifier ``__spam`` occurring in a class named ``Ham`` will be transformed
-to ``_Ham__spam``.  This transformation is independent of the syntactical
-context in which the identifier is used.  If the transformed name is extremely
-long (longer than 255 characters), implementation defined truncation may happen.
-If the class name consists only of underscores, no transformation is done.
+.. rubric:: Private name mangling
+
+When an identifier that textually occurs in a class definition begins with two
+or more underscore characters and does not end in two or more underscores, it
+is considered a :dfn:`private name` of that class.
```

Comment:
Maybe already mention this here?

```suggestion
is considered a :dfn:`private name` of that class. This doesn't make them
inaccessible, see below.
```

Reply:
I think this is quite obvious otherwise it does not make sense to declare such identifier (unless I say "inaccessible from the outside" but then it would stretch the introductory paragraph about the private name a bit too much IMO).